safe-global/safe-wallet-web
# feat: Show wallet balance

## What it solves

Displays the connected wallet balance in the header and in the tx-flow

[Figma Designs](https://www.figma.com/file/ptTs6lDBeUuLNySroJ5PiF/Web-Master-File?type=design&node-id=8647-11732&mode=design&t=25lotpzWMdjN7X2q-0) kudos to @TanyaEfremova !

## How this PR fixes it

- extracts the `useWallet` hook into a global context

## How to test it

1. Observe the wallet balance showing in the header for any connected wallet
2. Observe that it always reflects the connected wallet network not necessarily the network the safe is on
3. Create a transaction and observe seeing the wallet balance in the Execute block
4. Observe the new design according to the linked Figma

## Screenshots
<img width="602" alt="Screenshot 2023-10-23 at 09 20 39" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/12a4e16e-532d-4ed2-9740-97c05afe33e9">
<img width="697" alt="Screenshot 2023-10-23 at 09 20 58" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/f67499e7-4ecd-43da-ae61-8d5f82a0c547">
<img width="338" alt="Screenshot 2023-12-04 at 09 48 42" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/4e78435a-94db-4028-9137-d21fb0bf4054">
<img width="335" alt="Screenshot 2023-12-04 at 09 49 25" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/ffda8b44-d1e2-4c7d-a4de-e9824d5edb2b">
<img width="700" alt="Screenshot 2023-12-07 at 14 09 40" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/320b4c17-9951-4457-a177-1318cd436537">

## Checklist
